### DISCUSSION TITLE
How do I invoke Toast in the .ts/.js file (not in a svelte component?)

### DISCUSSION THREAD

**Discussion by Typogram** [skeletonlabs/skeleton#2492] [General]
error I got:
```
Uncaught (in promise) Error: Function called outside component initialization
```
it is a utility function that get called everywhere, so it is hard to capture the error in all the places in svlete and call the toast. I feel like it is a reasonably often use case to want to trigger Toast in .ts file.

**Comment by endigo9740** [comment]
@Typogram there's some known safety risk  using Svelte Stores server side. If these are files that do that, such as a load function, the error is very much intended as a safeguard.

This is part of why we have you initialize modals/toasts/drawers as described here:
https://www.skeleton.dev/utilities/toasts#initialize-stores

> This is required only once when implementing Skeleton's Drawer, Modal, and Toast stores and will prevent known issues with [SvelteKit SSR](https://github.com/skeletonlabs/skeleton/wiki/SvelteKit-SSR-Warning).

You can read more about the risk here:
https://github.com/sveltejs/kit/discussions/4339#discussioncomment-2384978

If these are scripts that run client-side, then I'd suggest keeping an eye on Svelte 5 progress. One of the key features is the introduction of  [Runes](https://svelte-5-preview.vercel.app/docs/runes), which are meant to allow for this sort of behavior.

**Reply by Typogram** [reply]
Thanks for the explaination @endigo9740 
In my case, I am trying to invoke the the toast from the client side (my app is svelte, not sveltekit) in the .ts file where there is a utility function to load an asset, there are times when the asset load failed for various reasons, and I would like to show a error toast to indicate that. However, limit to init store only in svelte file prevent me from doing that. 
Would it possible to enable this use case, as it doesn't have the safety risk related to using store in server side?

**Reply by endigo9740** [reply]
Just FYI we don't support Skeleton v2 outside of SvelteKit. Our stance is changing a bit for v3 though, as well as the technology that will power this. We'll be moving from Stores to State runes:

https://svelte-5-preview.vercel.app/docs/runes

One of the primary purposes of Rules is to enable Svelte features to work outside of `.svelte` components. Which is part of the reason we're right in the midst of rewriting everything for Skeleton v3.

So in short, it's not a request we can solve in the short term. But it should be resolved in v3 due to these technical changes coming down the pipeline.

**Reply by Typogram** [reply]
Thanks again! Excited for skeleton v3 support svelte 5 rune!
